src/library/Connect/LocalEnvironmentManager.cs

```diff
@@ -421,67 +421,69 @@ private void _ReportProgress(EventLevel eventLevel, string message, params objec
         /// </summary>
         public IDictionary<string, string> CreateEnvVariablesForK8s(WorkloadInfo workloadInfo)
         {
-            Dictionary<string, string> result = new Dictionary<string, string>(workloadInfo.EnvironmentVariables);
+            var result = new Dictionary<string, string>(workloadInfo.EnvironmentVariables);
+
             foreach (var endpoint in workloadInfo.ReachableEndpoints)
             {
-                if (StringComparer.OrdinalIgnoreCase.Equals(endpoint.DnsName, DAPR))
+                if (string.Equals(endpoint.DnsName, DAPR, StringComparison.OrdinalIgnoreCase))
                 {
                     // Override the DAPR env variables with the real local ports (that might be different if we neeeded to re-allocate them)
                     result["DAPR_HTTP_PORT"] = endpoint.Ports[0].LocalPort.ToString(); // TODO (lolodi): this is a hack that relies on the HTTP port to always be the first and GRPC port the second.
                     result["DAPR_GRPC_PORT"] = endpoint.Ports[1].LocalPort.ToString(); // We should probably name the port pairs (maybe with the env variable that we want to set with them).
-                }                                                                      // Once we do that, we can actually stop assigning the DAPR dns name ot this endpoint and just leave it empty, consistently with how the Remote agent works
+                                                                                       // Once we do that, we can actually stop assigning the DAPR dns name ot this endpoint and just leave it empty, consistently with how the Remote agent works
+                }
+
+                // because we are using dns name instead of service we have to retrieve it by splitting when needed
+                // If this ever cause issues we should consider larger refactor where we add serviceName member variable to EndpointInfo class.
+                var serviceName = endpoint.DnsName
+                    .ToUpperInvariant()
+                    .Split(".")
+                    .First()
+                    .Replace("-", "_")
+                    .Replace(".", "_");
+
+                var host = _useKubernetesServiceEnvironmentVariables || string.Equals(endpoint.DnsName, DAPR, StringComparison.OrdinalIgnoreCase)
+                    ? endpoint.LocalIP.ToString()
+                    : endpoint.DnsName;
+
+                if (string.Equals(serviceName, "KUBERNETES", StringComparison.OrdinalIgnoreCase))
+                    host = _kubernetesClient.HostName;

```

Comment:
nit: can you keep {} for styling with the rest of the code.

            if (serviceName == "KUBERNETES")
            {
                // reset KUBERNETES_SERVICE_HOST to cluster name.
                host = this._kubernetesClient.HostName;
            }

Reply:
Absolutely, will patch it in the upcoming commit.